test/utils/OrderPayable.sol.js

```diff
@@ -0,0 +1,319 @@
+const { ethers } = require('hardhat');
+const helpers = require('@nomicfoundation/hardhat-network-helpers');
+const { expect } = require('chai');
+const testUtils = require('../test-utils');
+
+describe('OrderPayable', function () {
+  async function deploy() {
+    const accounts = await ethers.getSigners();
+    const roles = {
+      deployer: accounts[0],
+      manager: accounts[1],
+      orderSigner: accounts[2],
+      withdrawer: accounts[3],
+      recipient: accounts[4],
+      randomPerson: accounts[9],
+    };
+
+    const AccessControlRegistry = await ethers.getContractFactory('AccessControlRegistry', roles.deployer);
+    const accessControlRegistry = await AccessControlRegistry.deploy();
+    await accessControlRegistry.deployed();
+
+    const adminRoleDescription = 'OrderPayable admin';
+
+    const OrderPayable = await ethers.getContractFactory('OrderPayable', roles.deployer);
+    const orderPayable = await OrderPayable.deploy(
+      accessControlRegistry.address,
+      adminRoleDescription,
+      roles.manager.address
+    );
+    await orderPayable.deployed();
+    return {
+      roles,
+      accessControlRegistry,
+      orderPayable,
+    };
+  }
+
+  describe('constructor', function () {
+    it('constructor', async function () {
+      const { roles, orderPayable } = await helpers.loadFixture(deploy);
+      expect(await orderPayable.manager()).to.equal(roles.manager.address);
+    });
+  });
+
+  describe('payForOrder', function () {
+    context('Order signer valid', function () {
+      context('Order Id not zero', function () {
```

Comment:
As a personal stylistic choice, I use regular English in `context()` strings. I'd prefer `Order signer is valid` (we don't need to shorten this as this is not a revert string) and `Order Id not zero` (this is not code, we don't need to use camel case).